test/mpdecimal_test.exs

```diff
@@ -2,27 +2,79 @@ defmodule MPDecimalTest do
   use ExUnit.Case
   doctest MPDecimal
 
-  test "correct digit handling" do
-    result = MPDecimal.Nif.power("3.14159", "1.58732")
-    assert result == "6.153686306332765322891030633"
+  test "positive integer exponents" do
+    assert_exp(2, 2, "4.0")
+    assert_exp(2, 3, "8.0")
+    assert_exp(1, 8, "1.0")
+    assert_exp(2, 8, "256.0")
+    assert_exp(2, 0, "1.0")
+    assert_exp(256, 0, "1.0")
   end
 
-  test "Strange NaN behavior???" do
-    result = MPDecimal.Nif.power("3.1", "1.587321")
-    assert result == "6.1536933506402224364507308462"
+  test "negative integer exponents" do
+    assert_exp(-1, 2, "1.0")
+    assert_exp(-2, 3, "-8.0")
+    assert_exp(-1, 0, "1.0")
+    assert_exp(-1, -1, "-1.0")
+    assert_exp(-2, -2, "0.25")
   end
 
-  test "bad args do not crash the VM" do
-    assert_raise ArgumentError, fn ->
-      MPDecimal.Nif.power("12.34", nil)
-    end
+  test "handling zero base values" do
+    assert_exp(0, 1, "0.0")
+    assert_exp(0, 2, "0.0")
 
-    assert_raise ArgumentError, fn ->
-      MPDecimal.Nif.power(false, "12.34")
-    end
+    assert_exp("0", 1, "0.0")
+    assert_exp("0", 2, "0.0")
+  end
+
+  test "decimal base values with integer powers" do
+    assert_exp("0.5", 2, "0.25")
+    assert_exp("0.5", 0, "1")
+    assert_exp("0.5", -1, "2")
+    assert_exp("0.5", -2, "4")
+    assert_exp("-0.5", 2, "0.25")
+    assert_exp("-0.5", 3, "-0.125")
+    assert_exp("-0.5", 0, "1")
+    assert_exp("-0.5", -1, "-2")
+    assert_exp("-0.5", -2, "4")
+    assert_exp("-0.5", -3, "-8")
+  end
+
+  test "integer base values with decimal powers" do
+    assert_exp(2, "2.5", "5.656854249492380195206754897")
+    assert_exp(2, "1.75", "3.363585661014858172124501905")
+    assert_exp(2, "0.9", "1.866065983073614831962686532")
+    assert_exp(2, "0.5", "1.414213562373095048801688724")
+    assert_exp(2, "0", "1")
+    assert_exp(2, "-0.5", "0.7071067811865475244008443621")
+    assert_exp(2, "-1.5", "0.3535533905932737622004221811")
+    assert_exp(0, "2.5", "0")
+    assert_exp(0, "1.75", "0")
+    assert_exp(0, "0.9", "0")
+    assert_exp(0, "0.5", "0")
+  end
+
+  test "infinity cases" do
+    assert_exp(0, -1, "Infinity")
```

Comment:
Currently we are raising an error for these use-cases (because we are using a float-based implementation), but I think the more correct result here is infinity. This matches the output that I get back from Wolfram Alpha